libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -183,15 +183,15 @@ string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const* _mess
 		return Whiskers(R"(
 			function <functionName>(condition <messageVars>) {
 				if iszero(condition) {
-					let fmp := mload(<freeMemPointer>)
-					mstore(fmp, <errorHash>)
-					let end := <abiEncodeFunc>(add(fmp, <hashHeaderSize>) <messageVars>)
-					revert(fmp, sub(end, fmp))
+					let memPtr := <allocateUnbounded>()
```

Comment:
Actually `allocateUnbounded` function is not performing the allocation but simply placing the `mload(fmp)` into a function called `allocateUnbounded`. To me, a function name such as `obtainFreeMemoryPtr` seems more appropriate.

Reply:
Also, the renaming of `fmp` to `memPtr` feels unnecessary since the semantics are unchanged.

Reply:
I think the idea here is that you should no longer be able to freely touch free memory pointer and should treat it as an implementation detail of the the allocate/release functions instead. `allocateUnbounded()` does not do anything with it currently but this might change.

Reply:
Sure, this is mostly perceived meaning. The main idea of this PR is to abstract away the free memory pointer in general. It now only occurs in the functions related to allocation.

Reply:
Note to self: This change is superficial.